core/shared/src/main/scala/hedgehog/package.scala

```diff
@@ -31,12 +31,15 @@ package object hedgehog extends ApplicativeSyntax {
   def propertyT: PropertyTOps =
     new PropertyTOps {}
 
-  implicit class Syntax[A](a1: A) {
+  implicit class Syntax[A](private val a1: A) extends AnyVal {
 
     // FIX Is there a way to get this to work with PropertyT and type-inference?
     def ====(a2: A): Result =
       Result.diffNamed("=== Not Equal ===", a1, a2)(_ == _)
 
-}
+    def matchPattern(right: PartialFunction[A, _]): Result =
```

Comment:
I really wish there was a super no-dependency, light-weight library _just_ for `Result`, this kind of matching is really it's own space where everyone has an opinion.

Reply:
Yeah I agree. I'm not so sure about this PR either as everyone may have an option just like you said.
So I just put something similar to what ScalaTest has https://www.scalatest.org/user_guide/using_matchers#matchingAPattern

Do you think it should be provided as a third-party extension? 🤔

Reply:
copying ScalaTest sounds good/fine to me. Just ignore me :)